Block/AbstractBlockService.php

```diff
@@ -22,6 +25,62 @@
 abstract class AbstractBlockService implements BlockServiceInterface
 {
     /**
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * @var EngineInterface
+     */
+    protected $templating;
+
+    /**
+     * @param string          $name
+     * @param EngineInterface $templating
+     */
+    public function __construct($name = null, EngineInterface $templating = null)
+    {
+        if (null === $name || null === $templating) {
```

Comment:
Can be shortened into `if (func_num_args() < 2)`, which might be a bit ugly, so do what you feel is best :P